server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -910,6 +951,77 @@ private static Settings resolveSettings(Metadata metadata, ComposableIndexTempla
+                                                  final ComposableIndexTemplate template,
+                                                  final IndicesService indicesService,
+                                                  final NamedXContentRegistry xContentRegistry) throws Exception {
+        final ClusterState stateWithTemplate = ClusterState.builder(state)
+            .metadata(Metadata.builder(state.metadata()).put(templateName, template))
+            .build();
+
+        Index createdIndex = null;
+        final String temporaryIndexName = "validate-template-" + UUIDs.randomBase64UUID().toLowerCase(Locale.ROOT);
+        try {
+            Settings resolvedSettings = resolveSettings(stateWithTemplate.metadata(), templateName);
+
+            // use the provided values, otherwise just pick valid dummy values
+            int dummyPartitionSize = IndexMetadata.INDEX_ROUTING_PARTITION_SIZE_SETTING.get(resolvedSettings);
+            int dummyShards = resolvedSettings.getAsInt(IndexMetadata.SETTING_NUMBER_OF_SHARDS,
+                dummyPartitionSize == 1 ? 1 : dummyPartitionSize + 1);
+            int shardReplicas = resolvedSettings.getAsInt(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0);
+
+
+            //create index service for parsing and validating "mappings"
+            Settings dummySettings = Settings.builder()
+                .put(IndexMetadata.SETTING_VERSION_CREATED, Version.CURRENT)
+                .put(resolvedSettings)
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, dummyShards)
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, shardReplicas)
+                .put(IndexMetadata.SETTING_INDEX_UUID, UUIDs.randomBase64UUID())
+                .build();
+
+            // Validate index metadata (settings)
+            final ClusterState stateWithIndex = ClusterState.builder(stateWithTemplate)
+                .metadata(Metadata.builder(stateWithTemplate.metadata())
+                    .put(IndexMetadata.builder(temporaryIndexName).settings(dummySettings))
+                    .build())
+                .build();
+            final IndexMetadata tmpIndexMetadata = stateWithIndex.metadata().index(temporaryIndexName);
+            IndexService dummyIndexService = indicesService.createIndex(tmpIndexMetadata, Collections.emptyList(), false);
+            createdIndex = dummyIndexService.index();
+
+            // Validate aliases
+            MetadataCreateIndexService.resolveAndValidateAliases(temporaryIndexName, Collections.emptySet(),
+                MetadataIndexTemplateService.resolveAliases(stateWithIndex.metadata(), templateName), stateWithIndex.metadata(),
+                new AliasValidator(),
+                // the context is only used for validation so it's fine to pass fake values for the
+                // shard id and the current timestamp
+                xContentRegistry, dummyIndexService.newQueryShardContext(0, null, () -> 0L, null));
```

Comment:
would using `indicesService.withTempIndexService` be better as it avoids adding.removing the index from the `IndexService` and also handles closing the `indexService`? (we're using this in `TransportSimulateIndexTemplateAction`)

Reply:
Yes absolutely, I'll change this